frontend/packages/knative-plugin/src/utils/create-eventsources-utils.ts

```diff
@@ -18,8 +19,34 @@ import * as camelSourceImg from '../imgs/logos/camelsource.svg';
 import * as containerSourceImg from '../imgs/logos/containersource.png';
 import * as cronJobSourceImg from '../imgs/logos/cronjobsource.png';
 import * as kafkaSourceImg from '../imgs/logos/kafkasource.svg';
+import CronJobSection from '../components/add/event-sources/CronJobSection';
+import SinkBindingSection from '../components/add/event-sources/SinkBindingSection';
+import ApiServerSection from '../components/add/event-sources/ApiServerSection';
+import YAMLEditorSection from '../components/add/event-sources/YAMLEditorSection';
+import KafkaSourceSection from '../components/add/event-sources/KafkaSourceSection';
 
-export const getEventSourcesDepResource = (formData: EventSourceFormData): K8sResourceKind => {
+const createResourceFromYAML = (formData: any): K8sResourceKind => {
+  const {
+    application: { name: applicationName },
+    yamlData,
+  } = formData;
+  const eventYaml = safeLoad(yamlData);
+  const { metadata } = eventYaml;
+  const defaultLabel = getAppLabels(metadata.name, applicationName);
+  const eventSourceResource: K8sResourceKind = {
+    ...eventYaml,
+    metadata: {
+      ...metadata,
+      labels: {
+        ...defaultLabel,
+      },
+      annotations,
+    },
+  };
+  return eventSourceResource;
+};
+
+const createResourceFromForm = (formData: any): K8sResourceKind => {
```

Comment:
This name doesn't make sense here since no actual resource is being created. Its just creating a data structure like the resource which is then used to create actual resource.

Reply:
renamed to `getResourceFromForm`